api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -730,4 +731,73 @@ public boolean isOrderTypeInUse(OrderType orderType) {
 		criteria.add(Restrictions.eq("encounter", encounter));
 		return criteria.list();
 	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrderGroupAttributeByUuid(String)
+	 */
+	@Override
+	public OrderGroupAttribute getOrderGroupAttributeByUuid(String uuid)  throws DAOException{
+		if(uuid==null||uuid.isEmpty()){
+			throw  new APIException("Uuid Cannot be Empty or null");
+		}
+	return (OrderGroupAttribute) sessionFactory.getCurrentSession().createQuery("from OrderGroupAttribute d where d.uuid = :uuid")
+			.setString("uuid", uuid).uniqueResult();
+	}
+
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getAllOrderGroupAttributeTypes()
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<OrderGroupAttributeType> getAllOrderGroupAttributeTypes() throws DAOException{
+		return sessionFactory.getCurrentSession().createCriteria(OrderGroupAttributeType.class).list();
+	}
+	
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrderGroupAttributeType(java.lang.Integer)
+	 */
+	@Override
+	public OrderGroupAttributeType getOrderGroupAttributeType(Integer id) throws DAOException{
+		return sessionFactory.getCurrentSession().get(OrderGroupAttributeType.class,id);
+	}
+	
+	/**
+	 * @see org.openmrs.api.db.OrderDAO#getOrderGroupAttributeTypeByUuid(java.lang.String)
+	 */
+	@Override
+	public OrderGroupAttributeType getOrderGroupAttributeTypeByUuid(String uuid) throws DAOException{
+		if(uuid==null||uuid.isEmpty()){
```

Comment:
Same as above

Reply:
@dkayiwa  cleaned out the spaces.

Reply:
I merged the commits which were done on master into branch TRUNK-5410.